File: 7-kyu/test's-results/python/solution.py
def test(r):
    average = round(sum(r) / len(r), 3)
    high_count = r.count(10) + r.count(9)
    avg_count = r.count(8) + r.count(7)
    low_count = len(r) - high_count - avg_count
    marks = {'h': high_count, 'a': avg_count, 'l': low_count}
    ret = [average, marks]
    if marks['a'] == 0 and marks['l'] == 0:
        ret.append('They did well')
    return ret

File: 7-kyu/test's-results/javascript/solution.js
const testResult = array => {
  const average = Math.round((array.reduce((acc, nxt) => acc + nxt, 0) / array.length) * 1000) / 1000;
  const dict = {h: 0, a: 0, l: 0};
  for (const mark of array) {
    if (mark < 7) {
      dict.l++;
    } else if (mark < 9) {
      dict.a++;
    } else {
      dict.h++;
    }
  }
  const result = [average, dict];
  if (array.every(mark => mark > 8)) {
    result.push("They did well");
  }
  return result;
};
